Lib/Search/EntityRepository.php

```diff
@@ -132,9 +132,8 @@ public function scanBy(array $criteria, $sizePerShard = 100, $expiryTime = '1m')
      */
     public function findOneBy(array $criteria)
     {
-        $options = array('field' => key($criteria));
-        $value = current($criteria);
-        return $this->_sm->getUnitOfWork()->load($this->_class, $value, $options);
+        $collection = $this->_sm->getUnitOfWork()->loadBy($this->_class, $criteria);
```

Comment:
[optional] loadBy takes $limit, where I probably will just give 1 to lighten the query.